### ISSUE TITLE
`SymLogNorm` and `SymLogScale` give inconsistent results....

### ISSUE THREAD

**Issue by jklymak** [matplotlib/matplotlib#16376]
on master....

```python
import matplotlib.pyplot as plt
import numpy as np
import matplotlib.scale as mscale
import matplotlib.colors as mcolors

#scale = mscale.SymmetricalLogScale(ax.xaxis, linthreshx=1.0)
fig, ax =plt.subplots()
ax.set_xscale('symlog', linthreshx=1.0, linscale=1)

ax.plot(np.arange(-100, 100, 0.1), np.arange(-100, 100, 0.1))
ax.set_xlim([-100, 100])
scale = ax.xaxis._scale

norm = mcolors.SymLogNorm(linthresh=1.0, linscale=1, vmin=-100, vmax=100)
xx = scale._transform.transform([-200, -100, -10, -1, -0.5, 0, 0.5, 1, 10, 100, 200])

print((xx - xx[1])/(xx[-2] - xx[1]))
print(norm([-200, -100, -10, -1, -0.5, 0, 0.5, 1, 10, 100, 200]))


fig, ax = plt.subplots()
ax.plot(norm([-200, -100, -10, -1, -0.5, 0, 0.5, 1, 10, 100, 200]), xx, '.')
ax.set_xlabel('Norm')
ax.set_ylabel('Scale')
```

yields

```
[-0.0483798207317112 0.0 0.1607142857142857 0.3214285714285714
 0.4107142857142857 0.5 0.5892857142857143 0.6785714285714286
 0.8392857142857143 1.0 1.0483798207317112]   # scale
[-0.05601509  0.          0.1860781   0.3721562   0.4360781   0.5
  0.5639219   0.6278438   0.8139219   1.          1.05601509]  # Norm
```

I think scale is right, and norm is wrong...

![NormIssue2](https://user-images.githubusercontent.com/1562854/73495747-06553d80-436c-11ea-8631-1544dc40c767.png)

![NormIssue](https://user-images.githubusercontent.com/1562854/73495719-f4739a80-436b-11ea-9051-b698af5e8925.png)

**Comment by jklymak** [comment]
Related to #16280 and #16286

**Comment by jklymak** [comment]
Marking release critical because anyone who uses `SymLogNorm` is possibly plotting corrupted data?

**Comment by jklymak** [comment]
I feel this issue is a pretty big emergency.  Anyone using this norm for published work is publishing incorrect data.  The fact that it’s been broken forever is pretty brutal because who knows how many papers have this error. (Hopefully few, because it’s a pretty unconventional norm).

**Comment by jklymak** [comment]
See https://github.com/matplotlib/matplotlib/issues/16375#issuecomment-580517570 for @anntzer possible path to fixing the problem.

**Comment by jklymak** [comment]
OK, the issue is that SymLogNorm is hard coded with base `np.e`, whereas scale takes a variable base, but defaults to 10.      I guess its arguable what the base should be by default. ping @dstansby?  Did you know the norm was not base-10, but rather base-e?  Should we just keep it that way for historic reasons?  I don't think most people want or expect base-e...

**Comment by jklymak** [comment]
Like the doc string even says "decades", which I don't  think it correct for base-e.

**Comment by dstansby** [comment]
I always read it as base 10, because it says 'decades'.

**Comment by dstansby** [comment]
PR open, thanks again for tracking this down and opening the issue

**Comment by dstansby** [comment]
Just a heads up that I'm going to try and tackle this again. https://github.com/matplotlib/matplotlib/pull/16391 as it stands has tests with hand-calculable values that I think for norm (without the changes in that PR) and scale.

**Comment by jklymak** [comment]
This was closed by #16457 FAICT...
